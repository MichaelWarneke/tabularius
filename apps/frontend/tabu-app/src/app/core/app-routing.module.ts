import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  {
    path: 'dash',
    loadChildren: '@tabularius/modules/app-dash#AppDashModule'
  },
  {
    path: 'journal',
    loadChildren: '@tabularius/modules/app-journal#AppJournalModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: []
})
export class AppRoutingModule {}
