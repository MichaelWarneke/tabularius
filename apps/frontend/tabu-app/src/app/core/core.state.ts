import {
  ActionReducerMap,
  MetaReducer,
  createFeatureSelector
} from '@ngrx/store';
import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { RouterStateUrl } from '@tabularius/shared/services';
import { storeFreeze } from 'ngrx-store-freeze';

import { initStateFromLocalStorageReducer } from '@tabularius/shared/services';
import { debugReducer } from '@tabularius/shared/services';

import { environment } from '../../environments/environment';

export interface AppState {
  router: RouterReducerState<RouterStateUrl>;
}

export const reducers: ActionReducerMap<AppState> = {
  router: routerReducer
};

export const metaReducers: MetaReducer<AppState>[] = [
  initStateFromLocalStorageReducer
];
if (!environment.production) {
  metaReducers.unshift(storeFreeze);
  metaReducers.unshift(debugReducer);
}

export const selectRouterState = createFeatureSelector<
  AppState,
  RouterReducerState<RouterStateUrl>
>('router');
