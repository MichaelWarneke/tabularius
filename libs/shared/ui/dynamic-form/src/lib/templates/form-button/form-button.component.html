<ng-container [formGroup]="group">
  <button
    class="form-button"
    mat-raised-button
    [color]="color"
    (click)="buttonClicked.emit()"
  >
    {{ text }}
  </button>
</ng-container>
