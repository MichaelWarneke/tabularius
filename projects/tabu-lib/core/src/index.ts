/*
 * Public API Surface of tabu-lib
 */

export * from './lib/core/core.module';
export * from './lib/material/material.module';


