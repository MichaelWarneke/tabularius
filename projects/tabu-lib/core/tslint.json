{
  "extends": "../../../tslint.json",
  "rules": {
    "directive-selector": [
      true,
      "attribute",
      "tabuLib",
      "camelCase"
    ],
    "component-selector": [
      true,
      "element",
      "tabu-lib",
      "kebab-case"
    ]
  }
}
