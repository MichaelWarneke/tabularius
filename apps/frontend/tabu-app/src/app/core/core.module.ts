import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppRoutingModule } from './app-routing.module';
import { TabuAppSidenavModule } from '@tabularius/ui';

@NgModule({
  declarations: [],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    TabuAppSidenavModule
  ],
  exports: [TabuAppSidenavModule]
})
export class CoreModule {}
