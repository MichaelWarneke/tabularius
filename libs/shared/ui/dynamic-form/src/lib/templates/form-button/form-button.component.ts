import { Component, Input, Output, EventEmitter } from '@angular/core';
import { FormGroup } from '@angular/forms';

@Component({
  selector: 'tabu-dynamic-form-button',
  templateUrl: './form-button.component.html',
  styleUrls: ['./form-button.component.scss']
})
export class FormButtonComponent {
  @Input() text: string | null = null;
  @Input() group: FormGroup | null = null;
  @Input() color: string | null = null;
  @Output() buttonClicked = new EventEmitter<any>();

  constructor() {}
}
