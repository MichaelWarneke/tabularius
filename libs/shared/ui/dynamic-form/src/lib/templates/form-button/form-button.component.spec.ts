import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { FormButtonComponent } from './form-button.component';
import { By } from '@angular/platform-browser';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('FormButtonComponent', () => {
  let component: FormButtonComponent;
  let fixture: ComponentFixture<FormButtonComponent>;
  const text = 'testText';
  const color = 'warn';

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [FormButtonComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(FormButtonComponent);
    component = fixture.componentInstance;
    component.text = text;
    component.color = color;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain the text', () => {
    const innerText = fixture.debugElement.query(By.css('button')).nativeElement
      .innerHTML;
    expect(innerText).toContain(text);
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
