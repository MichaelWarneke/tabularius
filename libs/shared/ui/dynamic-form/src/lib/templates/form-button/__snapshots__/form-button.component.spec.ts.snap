// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButtonComponent should render 1`] = `
<tabu-dynamic-form-button
  buttonClicked={[Function EventEmitter]}
  color={[Function String]}
  group="null"
  text={[Function String]}
>
  <button
    class="form-button"
    mat-raised-button=""
  >
     testText 
  </button>
</tabu-dynamic-form-button>
`;
